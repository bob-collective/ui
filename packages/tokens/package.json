{
  "name": "@gobob/tokens",
  "version": "0.0.0",
  "description": "Bob UI library.",
  "homepage": "https://github.com/bob-collective/bob-ui#readme",
  "license": "MIT",
  "keywords": [
    "bob",
    "tokens list"
  ],
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bob-collective/bob-ui.git",
    "directory": "packages/components"
  },
  "bugs": {
    "url": "https://github.com/bob-collective/bob-ui/issues"
  },
  "scripts": {
    "lint": "eslint .",
    "lint:fix": "eslint . --ext ts,tsx --fix",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@gobob/chains": "workspace:^",
    "@gobob/currency": "workspace:^",
    "tokenlist": "github:bob-collective/ethereum-optimism.github.io#f1cda28bfd05442182d387d9d1873ce039f2255c"
  }
}
