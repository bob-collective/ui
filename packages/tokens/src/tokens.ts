import { ChainId } from '@gobob/chains';

import { bobTokens, oldBobSepoliaTokens } from './constants/bob';
import { ethereumTokens } from './constants/eth';
import { sepoliaTokens } from './constants/sepolia';

export const tokens = {
  [ChainId.ETHEREUM]: ethereumTokens,
  [ChainId.SEPOLIA]: sepoliaTokens,
  [ChainId.BOB]: bobTokens,
  [ChainId.OLD_BOB_SEPOLIA]: oldBobSepoliaTokens,
  [ChainId.ARBITRUM_ONE]: [],
  [ChainId.BASE]: [],
  [ChainId.OP]: [],
  [ChainId.BSC]: [],
  [ChainId.OPBNB]: [],
  [ChainId.POLYGON]: [],
  [ChainId.POLYGON_ZKEVM]: [],
  [ChainId.MOONBEAM]: []
};
