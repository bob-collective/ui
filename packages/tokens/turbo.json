{
  "$schema": "https://turbo.build/schema.json",
  "extends": ["//"],
  "tasks": {
    "install": {
      "outputs": [],
      "cache": false
    }
  }
}
