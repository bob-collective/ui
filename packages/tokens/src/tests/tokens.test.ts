import { describe, it, expect } from 'vitest';

import { tokens } from '../tokens'; // Adjust the path as necessary

describe('Tokens Snapshot', () => {
  it('should match snapshot for tokens', () => {
    expect(tokens).toMatchSnapshot();
  });
});
