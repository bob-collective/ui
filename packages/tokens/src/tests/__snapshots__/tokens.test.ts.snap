// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tokens Snapshot > should match snapshot for tokens 1`] = `
{
  "1": {
    "usdc": ERC20Token {
      "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
      "chainId": 1,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "USD Coin",
      "projectLink": undefined,
      "symbol": "USDC",
    },
    "usdt": ERC20Token {
      "address": "0xdAC17F958D2ee523a2206206994597C13D831ec7",
      "chainId": 1,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "Tether USD",
      "projectLink": undefined,
      "symbol": "USDT",
    },
    "wbtc": ERC20Token {
      "address": "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
      "chainId": 1,
      "decimals": 8,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped BTC",
      "projectLink": undefined,
      "symbol": "WBTC",
    },
    "weth": ERC20Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "projectLink": "https://weth.io",
      "symbol": "WETH",
    },
  },
  "10": [],
  "1101": [],
  "11155111": {
    "usdc": ERC20Token {
      "address": "0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238",
      "chainId": 11155111,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "USDC",
      "projectLink": undefined,
      "symbol": "USDC",
    },
    "usdt": undefined,
    "wbtc": undefined,
    "weth": ERC20Token {
      "address": "0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9",
      "chainId": 11155111,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "projectLink": "https://weth.io",
      "symbol": "WETH",
    },
  },
  "1284": [],
  "137": [],
  "200901": [],
  "204": [],
  "4200": [],
  "42161": [],
  "56": [],
  "60808": {
    "tbtc": ERC20Token {
      "address": "0xBBa2eF945D523C4e2608C9E1214C2Cc64D4fc2e2",
      "chainId": 60808,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "tBTC v2",
      "projectLink": undefined,
      "symbol": "tBTC",
    },
    "usdc": ERC20Token {
      "address": "0xe75D0fB2C24A55cA1e3F96781a2bCC7bdba058F0",
      "chainId": 60808,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "Bridged USDC (BOB)",
      "projectLink": undefined,
      "symbol": "USDC",
    },
    "usdt": ERC20Token {
      "address": "0x05D032ac25d322df992303dCa074EE7392C117b9",
      "chainId": 60808,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "Tether USD",
      "projectLink": undefined,
      "symbol": "USDT",
    },
    "wbtc": ERC20Token {
      "address": "0x03C7054BCB39f7b2e5B2c7AcB37583e32D70Cfa3",
      "chainId": 60808,
      "decimals": 8,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped BTC",
      "projectLink": undefined,
      "symbol": "WBTC",
    },
    "weth": ERC20Token {
      "address": "0x0000000000000000000000000000000000000000",
      "chainId": 60808,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "projectLink": "https://weth.io",
      "symbol": "WETH",
    },
  },
  "808813": {
    "usdc": ERC20Token {
      "address": "0x2c1f73d37E7cdbA2792156139b0d35A765d595C5",
      "chainId": 808813,
      "decimals": 6,
      "isNative": false,
      "isToken": true,
      "name": "Bridged USDC (BOB)",
      "projectLink": undefined,
      "symbol": "USDC",
    },
    "weth": ERC20Token {
      "address": "0x4200000000000000000000000000000000000006",
      "chainId": 808813,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "projectLink": "https://weth.io",
      "symbol": "WETH",
    },
  },
  "8453": [],
}
`;
